test/functional/blockprod_generate_pos_blocks_rand_genesis_keys.py

```diff
@@ -0,0 +1,438 @@
+
+    def pack_transaction(self, transaction):
+        transaction_encoded = signed_transaction_obj.encode(transaction).to_hex()[2:]
+        transaction_id = ScaleBytes(
+            mintlayer_hash(transaction_obj.encode(transaction["transaction"]).data)
+        ).to_hex()[2:]
+
+        return (transaction_encoded, transaction_id)
+
+    def previous_block_id(self):
+        previous_block_id = self.nodes[0].chainstate_best_block_id()
+        return self.hex_to_dec_array(previous_block_id)
+
+    def stake_private_key(self, stake_private_key):
+        return {
+            "key": {
+                "Secp256k1Schnorr": {
+                    "data": self.hex_to_dec_array(stake_private_key),
+                },
+            },
+        }
+
+    def stake_public_key(self, stake_public_key):
+        return {
+            "key": {
+                "Secp256k1Schnorr": {
+                    "pubkey_data": self.hex_to_dec_array(stake_public_key),
+                },
+            },
+        }
+
+    def vrf_private_key(self, vrf_private_key):
+        return {
+            "key": {
+                "Schnorrkel": {
+                    "key": self.hex_to_dec_array(vrf_private_key),
+                },
+            },
+        }
+
+    def vrf_public_key(self, vrf_public_key):
+        return {
+            "key": {
+                "Schnorrkel": {
+                    "key": self.hex_to_dec_array(vrf_public_key),
+                },
+            },
+        }
+
+    def run_test(self):
+        global genesis_stake_private_key_hex, genesis_stake_public_key_hex
+        genesis_stake_private_key = self.stake_private_key(genesis_stake_private_key_hex[2:])
+        genesis_stake_public_key = self.stake_public_key(genesis_stake_public_key_hex[2:])
+
+        global genesis_vrf_private_key_hex, genesis_vrf_public_key_hex
+        genesis_vrf_private_key = self.vrf_private_key(genesis_vrf_private_key_hex[2:])
+        genesis_vrf_public_key = self.vrf_public_key(genesis_vrf_public_key_hex[2:])
+
+        #
+        # Transfer Genesis UTXO to AnyoneCanSpend
+        #
+
+        block_input_data = block_input_data_obj.encode({
+            "PoS": {
+                "stake_private_key": genesis_stake_private_key,
+                "vrf_private_key": genesis_vrf_private_key,
+                "pool_id": genesis_pool_id,
+                "kernel_inputs": [
+                        {
+                            "Utxo": {
+                                "id": {
+                                    "BlockReward": self.previous_block_id()
+                                },
+                                "index": 1,
+                            },
+                        },
+                ],
+                 "kernel_input_utxo": [
+                    {
+                        "CreateStakePool": [
+                            genesis_pool_id,
+                            {
+                                "value": "4000000000000000",
```

Comment:
Perhaps we should find a proper way to calculate coins from atoms instead this chaotic zeros' situation.

Reply:
I've added it as an item to #1013